### ISSUE TITLE
CombinedLoss.forward() swaps estimate/target when routing into asr_loss

### ISSUE THREAD

**Issue by udsy19** [NVIDIA-NeMo/NeMo#16168]
**Describe the bug**

`CombinedLoss.forward()` in `nemo/collections/audio/losses/maxine/losses_combined.py` binds its two
locals backwards before calling `asr_loss`:

```python
primary_audio = estimate[..., :min_len]
predicted_audio = target[..., :min_len]
```

Everywhere else in the file (`spectral_loss(self, predicted_audio, primary_audio)`,
`asr_loss(self, predicted_audio, primary_audio)`) and at the sole production call site
(`BNR2Model._compute_train_loss`, `nemo/collections/audio/models/maxine/bnr.py:222-224`, which names
the model's own output `predicted_audio` and passes it as `estimate=` to `CombinedLoss`), the naming
convention is unambiguous: `predicted_audio` is the model's own output, `primary_audio` is the
ground-truth reference. The swap means `asr_loss(predicted_audio, primary_audio)` actually receives
the *target* audio's ASR posterior as `predicted_audio` and the *estimate*'s posterior as
`primary_audio` — exactly backwards.

Inside `asr_loss`, both signals are passed through the same frozen ASR model, then
`torch.nn.CrossEntropyLoss()(predicted_prob, primary_prob)` is computed. With the swap, the clean
reference audio's posterior is used where the model's own (trainable) estimate should be, and vice
versa — backwards from the intended objective of pulling the enhanced signal's ASR posterior toward
the clean signal's.

**Steps/Code to reproduce bug**

```python
import torch
from nemo.collections.audio.losses.maxine import CombinedLoss

loss_fn = CombinedLoss(
    sample_rate=16000, fft_length=64, hop_length=16, num_mels=8,
    sisnr_loss_weight=0.0, spectral_loss_weight=0.0, asr_loss_weight=1.0,
    use_asr_loss=False, use_mel_spec=False,
)
loss_fn.dummy_param = torch.nn.Parameter(torch.zeros(1))
loss_fn.use_asr_loss = True

recorded = {}
def fake_asr_loss(self, predicted_audio, primary_audio):
    recorded['predicted_audio'] = predicted_audio.clone()
    recorded['primary_audio'] = primary_audio.clone()
    return torch.tensor(0.0)
loss_fn.asr_loss = fake_asr_loss.__get__(loss_fn, CombinedLoss)

torch.manual_seed(0)
estimate = torch.randn(2, 1, 320)
target = torch.randn(2, 1, 320) + 5.0
loss_fn.forward(estimate=estimate, target=target)

print(torch.equal(recorded['predicted_audio'], estimate))  # prints False on current main
print(torch.equal(recorded['primary_audio'], target))      # prints False on current main
```

Also confirmed with the real cached `stt_en_conformer_ctc_small` checkpoint via the existing
`tests/collections/audio/test_audio_losses.py::TestAudioLosses::test_maxine_combined_loss` golden
values: fixing the routing changes the ASR-only config's loss from `1.9749` to `1.9083` (and the
combined config from `101.1941` to `101.1275` by the same delta), while the SiSNR-only and
spectral-only configs are unchanged — both of those losses are provably symmetric under this
particular swap, so only the ASR branch is affected.

**Expected behavior**

`asr_loss`'s `predicted_audio` argument should receive the model's own estimate (the differentiable
signal), and `primary_audio` should receive the ground-truth target — matching every other call site
and the file's own naming convention.

**Environment overview**

- Environment location: Bare-metal Linux
- Method of NeMo install: from a `git` checkout of `NVIDIA-NeMo/Speech` at commit `6281c939aa`

**Environment details**

- OS: Ubuntu 24.04.4 LTS, kernel 6.8.0-138-generic
- PyTorch: 2.13.0+cpu
- Python: 3.12.3

**Additional context**

`git log -S` on the swapped assignment shows it has been present, untouched, since the line was
introduced in #12655 ("Add Maxine BNR 2.0 network implementation to NeMo") — no later commit revisited
it, and no test in the tree asserts this routing as intentional. This is a shipped, documented training
path: `examples/audio/conf/maxine_bnr.yaml` sets `loss._target_: ...CombinedLoss` with
`use_asr_loss: true` by default.
